spec.html

```diff
@@ -8468,82 +8468,33 @@ <h1>[[Construct]] ( _argumentsList_, _newTarget_ )</h1>
       </emu-alg>
     </emu-clause>
 
-    <emu-clause id="sec-functionallocate" aoid="FunctionAllocate">
-      <h1>FunctionAllocate ( _functionPrototype_ )</h1>
-      <p>The abstract operation FunctionAllocate requires the argument _functionPrototype_. FunctionAllocate performs the following steps:</p>
+    <emu-clause id="sec-ordinaryfunctioncreate" aoid="OrdinaryFunctionCreate" oldids="sec-functionallocate,sec-functioninitialize,sec-functioncreate,sec-generatorfunctioncreate,sec-asyncgeneratorfunctioncreate,sec-async-functions-abstract-operations-async-function-create">
+      <h1>OrdinaryFunctionCreate ( _functionPrototype_, _ParameterList_, _Body_, _thisMode_, _Scope_ )</h1>
```

Comment:
What's up with the capitalization here for the parameter names anyways?

Reply:
Back in ES3, "13.2 Creating Function Objects" says:
> Given an optional parameter list specified by _FormalParameterList_, a body specified by _FunctionBody_, and a scope chain specified by _Scope_, a Function object is constructed as follows: ...

Here, the _FormalParameterList_ and _FunctionBody_ were nonterminals (in the productions for _FunctionDeclaration_ and _FunctionExpression_), so were necessarily capitalized, as that's always been the convention in ES grammars. _Scope_ wasn't a nonterminal, so didn't *need* to be capitalized, but on the other hand ES3 had very few aliases/metavariables/parameters, and there wasn't a strong convention either for or against capitalization. So maybe it got capitalized just to look like  _FormalParameterList_ and _FunctionBody_.

ES5 added tons of aliases/metavariables/parameters, with a fairly strong convention for _camelCase_, but "13.2 Creating Function Objects" retained ES3's capitalization, and added another capitalized parameter:
> Given an optional parameter list specified by _FormalParameterList_, a body specified by _FunctionBody_, a Lexical Environment specified by _Scope_, and a Boolean flag _Strict_, a Function object is constructed as follows: ...

Here (based on looking at invocations), each of _FormalParameterList_ and _FunctionBody_ was a fuzzy combination of nonterminal and parameter.

ES6 converted this to a proper abstract operation: `FunctionCreate (kind, ParameterList, Body, Scope, Strict, prototype)`:
> The abstract operation FunctionCreate requires the arguments: _kind_ which is one of (Normal, Method, Arrow), a parameter list production specified by _ParameterList_, a body production specified by _Body_, a Lexical Environment specified by _Scope_, a Boolean flag _Strict_, and optionally, an object _prototype_. FunctionCreate performs the following steps: ...

Here, _ParameterList_ and _Body_ are definitely just parameters, not nonterminals, but they retain the capitalization of their earlier analogs. Likewise, the capitalization of _Scope_ and _Strict_ is simply carried forward.

So the capitalization was originally unremarkable, and then, over time, the surrounding spec changed in ways that rendered it anomalous.

See also issue #1749.